awx/main/models/credential/__init__.py

```diff
@@ -461,8 +480,18 @@ def get_input(self, field_name, **kwargs):
         raise AttributeError(field_name)
 
     def has_input(self, field_name):
+        if field_name in self.dynamic_input_fields:
+            return True
         return field_name in self.inputs and self.inputs[field_name] not in ('', None)
 
+    def _get_dynamic_input_field_names(self):
+        input_sources = CredentialInputSource.objects.filter(target_credential=self)
+        return [obj.input_field_name for obj in input_sources]
+
+    def _get_dynamic_input(self, field_name):
+        input_sources = CredentialInputSource.objects.filter(target_credential=self)
+        return input_sources.filter(input_field_name=field_name).first().get_input_value()
```

Comment:
This is an extra query, and it surfaces errors in a confusing way ("NoneType has no attribute `get_input_value`").

```python
for input_source in self.input_sources.all():
    if input_source.input_field_name == field_name:
        return input_source.get_input_value()
else:
    raise RuntimeError('Field name {} is not a dynamic input field, choices are {}...")
```

I just want to throw that out there, and I know it feels like more code, but it avoids extra queries if you can go through the effort to prefetch the `input_sources`.